x/ccv/provider/keeper/misbehaviour.go

```diff
@@ -142,8 +142,20 @@ func headerToLightBlock(h ibctmtypes.Header) (*tmtypes.LightBlock, error) {
 }
 
 // CheckMisbehaviour checks that headers in the given misbehaviour forms
-// a valid light client attack and that the corresponding light client isn't expired
+// a valid light client attack from an ICS consumer chain and that the light client isn't expired
 func (k Keeper) CheckMisbehaviour(ctx sdk.Context, misbehaviour ibctmtypes.Misbehaviour) error {
+	// check that the misbehaviour is for an ICS consumer chain
+	clientID, found := k.GetConsumerClientId(ctx, misbehaviour.Header1.Header.ChainID)
+	if !found {
+		return fmt.Errorf("incorrect misbehaviour with conflicting headers from a non-existent consumer chain: %s", misbehaviour.Header1.Header.ChainID)
+	} else if misbehaviour.ClientId != clientID {
+		return fmt.Errorf("incorrect misbehaviour: expected client ID for consumer chain %s is %s got %s",
+			misbehaviour.Header1.Header.ChainID,
+			clientID,
+			misbehaviour.ClientId,
+		)
+	}
+
 	clientState, found := k.clientKeeper.GetClientState(ctx, misbehaviour.GetClientID())
```

Comment:
nit: We could use directly the `clientId` here. 
```suggestion
	clientState, found := k.clientKeeper.GetClientState(ctx, clientId)
```
This way we would avoid a potential regression where the previous `else if misbehaviour.ClientId != clientID` statement is accidentally removed and `misbehaviour.GetClientID()` is something else.

Reply:
Or it might make sense to add some sanity tests for this to prevent a possible regression.